src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -114,7 +115,8 @@
    :version s/Str
    :environment (s/maybe s/Str)
    :transaction_uuid (s/maybe s/Str)
-   :producer_timestamp (s/maybe pls/Timestamp)
+   :producer_timestamp (s/pred #(not (nil? (to-timestamp %))) 'coercable-to-timestamp)
```

Comment:
only question I have is whether we want/need this. My understanding is it's not needed and could just be pls/Timestamp, in which case it seems pretty heavy to keep around.